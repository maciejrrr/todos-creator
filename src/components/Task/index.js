import React from 'react';
import styled from 'styled-components';
import { Draggable } from 'react-beautiful-dnd';

import { ItemTypes } from '../../containers/Card/constants';
import { colors, metrics } from '../../theme';

const TaskContainer = styled.div`
  background-color: ${colors.white};
  padding: ${metrics.smallMargin}px;
  box-shadow: ${colors.boxShadow};
  margin-top: ${metrics.baseMargin}px;
  color: ${colors.darkGrayText};
  :last-child {
    margin-bottom: 0;
  }
`;

const TaskName = styled.p`
  word-wrap: break-word;
`;

const Task = ({ task }) => {
  const { id, name } = task;
  return (
    <Draggable draggableId={id} type={ItemTypes.TASK}>
      {(provided, snapshot) => (
        <div>
          <TaskContainer
            innerRef={provided.innerRef}
            style={provided.draggableStyle}
            {...provided.dragHandleProps}
          >
            <TaskName>{name}</TaskName>
          </TaskContainer>
          {provided.placeholder}
        </div>
      )}
    </Draggable>
  );
};

export default Task;
