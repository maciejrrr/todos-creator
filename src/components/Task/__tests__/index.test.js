import Task from '../index';
import { DragDropContext } from 'react-beautiful-dnd';

describe('Task component', () => {
  let props;
  beforeEach(() => {
    props = {
      task: {
        id: 1,
        name: 'test task',
      },
    };
  });

  describe('snapshots', () => {
    it('task with passed name', () => {
      const tree = renderer
        .create(
          <DragDropContext>
            <Task {...props} />
          </DragDropContext>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
