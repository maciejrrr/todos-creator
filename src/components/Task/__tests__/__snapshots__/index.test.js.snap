// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component snapshots task with passed name 1`] = `
<div>
  <div
    aria-grabbed={false}
    className="sc-bdVaJa hlCdMR"
    draggable={false}
    onClick={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "pointerEvents": "auto",
        "transform": null,
        "transition": null,
      }
    }
    tabIndex={0}
  >
    <p
      className="sc-bwzfXH dihBbR"
    >
      test task
    </p>
  </div>
</div>
`;
